src/server/controllers/usersController.js

```diff
@@ -20,138 +20,139 @@ const createUser = asyncHandler(async (req, res) => {
-    res.status(201).json({ message: "User successfully created!", id: newUser._id });
-  } else {
-    res.status(400).json({ message: "User unsuccessfully created." });
-  }
-});
+    // Hashing passwords to encrypt user data
+    const hashPwd = await bcrypt.hash(password, 10);
+    const userDocument = { firstname, lastname, email, password: hashPwd };
+    const newUser = await User.create(userDocument);
 
-const getUserByEmail = async (req, res) => {
-  const user = await User.findOne({ email: req.body.email }).then((user) => {
-    if (user) {
-      console.log(`Found user ${user.email}`);
-      res.status(200).json(user);
+    if (newUser) {
+        res.status(201).json({ message: "User successfully created!", id: newUser._id });
     } else {
-      return res.status(400).json({ message: "No user found" });
+        res.status(400).json({ message: "User unsuccessfully created." });
     }
-  });
+});
+
+const getUserByEmail = async(req, res) => {
+    const user = await User.findOne({ email: req.body.email }).then((user) => {
+        if (user) {
+            console.log(`Found user ${user.email}`);
+            res.status(200).json(user);
+        } else {
+            return res.status(400).json({ message: "No user found" });
+        }
+    });
 
 };
 
 // This action is to verify the credentials of the user when logging in
-const verifyUser = async (req, res) => {
-  const user = await User.findOne({
-    email: req.body.email,
-    password: req.body.password,
-  }).then((user) => {
-    if (user) {
-      req.session.email = user.email;
-      console.log(`Found user ${user.email}`);
-      res.status(200).json(user);
+const verifyUser = async(req, res) => {
+    const { email, password } = req.body;
+    const user = await User.findOne({ email }).exec();
+
+    if (!user) {
+        return res.status(404).json({ message: 'User not found' });
+    }
+
+    const passwordMatches = await bcrypt.compare(password, user.password);
+
+    if (passwordMatches) {
+        req.session.email = user.email;
+        console.log(`Found user ${user.email}`);
+        res.status(200).json(user);
     } else {
-      return res.status(404).json({
-        errors: [{ user: "not found" }],
-      });
+        return res.status(401).json({ message: 'Incorrect password' });
     }
-  });
 };
 
 //This will be used to update the user's profile information
-const updateUser = async (req, res) => {
-  
+const updateUser = async(req, res) => {
+
 };
 //Action of deleting  a user from awaiting connections list 
-const deleteAwaitingConnections  = async (req, res) => {
-  const { firstname , lastname} = req.body;
-  const  _id = "63f41b0123e995b64434ece0";
-  const user = await User.findOneAndUpdate(
-    {_id: _id},
-    {$pull: { waitingConnections: {firstname: firstname, lastname: lastname}
-    }})
+const deleteAwaitingConnections = async(req, res) => {
+    const { firstname, lastname } = req.body;
+    const _id = "63f41b0123e995b64434ece0";
```

Comment:
Once we've implemented the user session feature, let's remove the hard coded IDs to make these functions work for all users.